src/proxy/connection_manager.rs

```diff
@@ -260,6 +260,39 @@ impl ConnectionManager {
         // potentially large copy under read lock, could require optimization
         self.drains.read().expect("mutex").keys().cloned().collect()
     }
+
+    /// Close all inbound and outbound connections
+    /// This is called when certificate revocations are detected
+    pub fn close_all_connections(&self) {
```

Comment:
This seems really dangerous and bad behavior. How strongly do you feel about keeping this?

Does any other application do this?

Reply:
The only reason I addd this since http connections are long lived and does not check updated crl. Is there a better way to handle this? I know envoy has some timeout we can configure. Not comepletly sure how zTunnel handles it. Happy to twek this.

Reply:
My intuition is to do nothing. But at the very least, only close ones that are revoked

Reply:
Doing nothing feels wrong IMO; the purpose of a CRL is to be able to break a particular trust chain, and you don't want existing connections to continue. Only closing the ones that are revoked is the right solution to me

Reply:
Complexity aside, is there a strong argument we can make for preserving outbound connections which wouldn't presently be allowed due to revoked certs?

Reply:
Looks like [OpenVPN](https://openvpn.net/community-docs/revoking-certificates.html) forces a reboot:

>"...and have the new CRL take effect immediately for newly connecting clients. If the client whose certificate you are revoking is already connected, you can restart the server via a signal (SIGUSR1 or SIGHUP) and flush all clients, or you can telnet to the management interface and explicitly kill the specific client..."

Reply:
Is restart your clients if you want the CRL to take effect quickly the UX we want then?

Reply:
Separate PR seems reasonable to me so we can evaluate the complexity on its own